### ISSUE TITLE
Primordia: saved games from older build cannot be loaded

### ISSUE THREAD

**Issue by rathann** [adventuregamestudio/ags#1672]
**Describe the bug**
Saved games from older Primordia build (GOG 8 Feb 2016) cannot be loaded in newer build (GOG 17 Oct 2019).

**AGS Version**
ags-3.6.0.26 (reproducible in 3.5.0.32 and with the 3.5.0.26 bundled with the newer Primordia build as well)

**Game**
Primordia GOG build 17 Oct 2019.

**To Reproduce**
1. Start the game
2. Click on LOAD GAME
3. Select any old saved game

**Actual behavior**
```
Adventure Game Studio v3.6 Interpreter
Copyright (c) 1999-2011 Chris Jones and 2011-2022 others
ACI version 3.6.0.26

Initializing backend libs
Initializing game data
Opened game data file: game28.dta
Game data version: 50
Compiled with: 3.5.0.26
Startup directory: /usr/share/primordia/
Data directory: /usr/share/primordia/
Setting up game configuration
Voice pack found: speech.vox
audio.vox found and initialized.
Initializing TTF renderer
Initializing mouse: number of buttons reported is 3
Install timer
Audio driver: pulseaudio
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
AudioCore: opened device "Built-in Audio Analog Stereo"
Supported sound decoders:
 - Play modules through ModPlug : 669,AMF,AMS,DBM,DMF,DSM,FAR,GDM,IT,MDL,MED,MOD,MT2,MTM,OKT,PTM,PSM,S3M,STM,ULT,UMX,XM,
 - MPEG-1 Audio Layer I-III : MP3,MP2,MP1,
 - Microsoft WAVE audio format : WAV,
 - Audio Interchange File Format : AIFF,AIF,
 - Sun/NeXT audio file format : AU,
 - Ogg Vorbis audio : OGG,
 - Creative Labs Voice format : VOC,
 - Raw audio : RAW,
 - Shorten-compressed audio data : SHN,
 - Free Lossless Audio Codec : FLAC,FLA,
Install exit handler
Initialize path finder library
Game title: 'Primordia'
Game GUI version: 119
Requested script API: v3.5.0-final (7), compat level: v3.5.0-final (7)
WARNING: font 'agsfnt0.wfn' has mistakes in data format, some characters may be displayed incorrectly
WARNING: font 'agsfnt3.wfn' has mistakes in data format, some characters may be displayed incorrectly
WARNING: font 'agsfnt5.wfn' has mistakes in data format, some characters may be displayed incorrectly
WARNING: font 'agsfnt16.wfn' has mistakes in data format, some characters may be displayed incorrectly
WARNING: font 'agsfnt19.wfn' has mistakes in data format, some characters may be displayed incorrectly
WARNING: font 'agsfnt23.wfn' has mistakes in data format, some characters may be displayed incorrectly
WARNING: font 'agsfnt26.wfn' has mistakes in data format, some characters may be displayed incorrectly
Plugin 'agsteam' could not be loaded (expected 'libagsteam.so'), trying built-in plugins...
Placeholder functions for the plugin 'agsteam' found.
Checking for disk space
Game native resolution: 320 x 200 (32 bit)
Graphic settings: driver: D3D9, windowed: yes, screen size: 0 x 0, game scale: round
Requested graphics driver 'D3D9' not found, will try existing drivers instead
Running OpenGL: 3.1
Mouse speed control: enabled, unit: 1.000000, user value: 1.000000
Setting up window
Initialize sprites
Engine initialization complete
Starting game
WARNING: the saved game '/home/rathann/.local/share/ags/Primordia/agssave.021' references game file 'agsgame.dat' (title: ''), but it cannot be found in the current directory. Trying to restore in the running game instead.
Unable to restore the saved game.
Saved content does not match current game.
Mismatching size of global script data.
An internal error has occurred. Please note down the following information.
If the problem persists, contact the game author for support or post these details on the AGS Technical Forum.
(ACI version 3.6.0.26)

Error: Unable to restore the saved game.
Saved content does not match current game.
Mismatching size of global script data.

***** ENGINE HAS SHUTDOWN
```

**Expected behavior**
Saved game should be loaded successfully.

**Desktop (please complete the following information):**
 - OS: Fedora Linux
 - Version: 36

**Comment by ericoporto** [comment]
@Dualnames1

**Comment by ivan-mogilko** [comment]
> Saved content does not match current game.
Mismatching size of global script data.

This is a common problem with AGS: its save format does not allow to restore the saves if the game contents changed in certain way.

More detailed information can be found here, although its mostly interesting for the game developers:
https://github.com/adventuregamestudio/ags-manual/wiki/GameSavesCompatibility

**Comment by Dualnames1** [comment]
There is a way to grab an older version from gog. If you go to configure, I remember there is some sort of client that allows that, donn't remember which. Regardless, rathann, if you wish to at any point explain what you're trying to accomplish here, I'll be able to help. And this is a Primordia issue, not an engine issue.

**Comment by rathann** [comment]
@Dualnames1 thanks, but I'm still out of luck. I understand this is a Primordia issue (or, rather, a limitation of the game engine which cannot be fixed for existing games and saved states). I made these saves using an old Linux build of Primordia and I'm trying to continue playing where I left off in November 2016. Unfortunately, all of the builds that I'm able to download using https://github.com/Sude-/lgogdownloader fail with the same error. It's only possible to grab old Windows and Mac builds using either lgogdownloader or GOG Galaxy and there seems to be no way to download old Linux builds. And yes, I've opened a support ticket with GOG already.
